External/diffmatchpatch/DiffMatchPatchCFUtilities.c

```diff
@@ -648,27 +655,19 @@ CFIndex diff_cleanupSemanticScore(CFStringRef one, CFStringRef two) {
   // 'whitespace'.  Since this function's purpose is largely cosmetic,
   // the choice has been made to use each language's native features
   // rather than force total conformity.
-  UniChar char1 =
-  CFStringGetCharacterAtIndex(one, (CFStringGetLength(one) - 1));
-  UniChar char2 =
-  CFStringGetCharacterAtIndex(two, 0);
-  Boolean nonAlphaNumeric1 =
-  !CFCharacterSetIsCharacterMember(alphaNumericSet, char1);
-  Boolean nonAlphaNumeric2 =
-  !CFCharacterSetIsCharacterMember(alphaNumericSet, char2);
-  Boolean whitespace1 =
-  nonAlphaNumeric1 && CFCharacterSetIsCharacterMember(whiteSpaceSet, char1);
-  Boolean whitespace2 =
-  nonAlphaNumeric2 && CFCharacterSetIsCharacterMember(whiteSpaceSet, char2);
-  Boolean lineBreak1 =
-  whitespace1 && CFCharacterSetIsCharacterMember(controlSet, char1);
-  Boolean lineBreak2 =
-  whitespace2 && CFCharacterSetIsCharacterMember(controlSet, char2);
-  Boolean blankLine1 =
-  lineBreak1 && diff_regExMatch(one, &blankLineEndRegEx);
-  Boolean blankLine2 =
-  lineBreak2 && diff_regExMatch(two, &blankLineStartRegEx);
-  
+  UniChar char1 = CFStringGetCharacterAtIndex(one, (CFStringGetLength(one) - 1));
+  UniChar char2 = CFStringGetCharacterAtIndex(two, 0);
+  Boolean char1IsSurrogate = CFStringIsSurrogateLowCharacter(char1) || CFStringIsSurrogateHighCharacter(char1);
+  Boolean char2IsSurrogate = CFStringIsSurrogateLowCharacter(char2) || CFStringIsSurrogateHighCharacter(char1);
+  Boolean nonAlphaNumeric1 = !CFCharacterSetIsCharacterMember(alphaNumericSet, char1);
+  Boolean nonAlphaNumeric2 = !CFCharacterSetIsCharacterMember(alphaNumericSet, char2);
+  Boolean whitespace1 = nonAlphaNumeric1 && CFCharacterSetIsCharacterMember(whiteSpaceSet, char1);
+  Boolean whitespace2 = nonAlphaNumeric2 && CFCharacterSetIsCharacterMember(whiteSpaceSet, char2);
+  Boolean lineBreak1 = whitespace1 && CFCharacterSetIsCharacterMember(controlSet, char1);
+  Boolean lineBreak2 = whitespace2 && CFCharacterSetIsCharacterMember(controlSet, char2);
+  Boolean blankLine1 = lineBreak1 && diff_regExMatch(one, &blankLineEndRegEx);
+  Boolean blankLine2 = lineBreak2 && diff_regExMatch(two, &blankLineStartRegEx);
+
```

Comment:
Can we remove the whitespace changes? It's making it hard to scan and see if any actual changes occurred here.

Reply:
Only two new lines there... 

```
  Boolean char1IsSurrogate = CFStringIsSurrogateLowCharacter(char1) || CFStringIsSurrogateHighCharacter(char1);
   Boolean char2IsSurrogate = CFStringIsSurrogateLowCharacter(char2) || CFStringIsSurrogateHighCharacter(char1);
```

I've relocated all the things because it was **next to impossible** to understand what was going on. Lemme see if i can revert that... relatively easily!

Reply:
@dmsnell there you go! The changes themselves are **very** low footprint. And the beauty of it? if there are no surrogate pairs anywhere... this will run the same way it ever did.